pkg/daemon/daemon_test.go

```diff
@@ -0,0 +1,265 @@
+					Interfaces: []sriovnetworkv1.InterfaceExt{
+						{
+							VFs: []sriovnetworkv1.VirtualFunction{
+								{},
+							},
+							DeviceID:   "158b",
+							Driver:     "i40e",
+							Mtu:        1500,
+							Name:       "ens803f0",
+							PciAddress: "0000:86:00.0",
+							Vendor:     "8086",
+							NumVfs:     4,
+							TotalVfs:   64,
+						},
+					},
+				},
+			}
+			Expect(
+				createSriovNetworkNodeState(sut.client, nodeState)).
+				To(BeNil())
+
+			var msg Message
+			Eventually(refreshCh, "10s").Should(Receive(&msg))
+			Expect(msg.syncStatus).To(Equal("InProgress"))
+
+			Eventually(refreshCh, "10s").Should(Receive(&msg))
+			Expect(msg.syncStatus).To(Equal("Succeeded"))
+
+			Eventually(func() (int, error) {
+				podList, err := sut.kubeClient.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{
+					LabelSelector: "app=sriov-device-plugin",
+					FieldSelector: "spec.nodeName=test-node",
+				})
+
+				if err != nil {
+					return 0, err
+				}
+
+				return len(podList.Items), nil
+			}, "10s").Should(BeZero())
+
+		})
+
+		It("ignore non latest SriovNetworkNodeState generations", func() {
+			go func() {
+				Expect(sut.Run(stopCh, exitCh)).To(BeNil())
+			}()
+
+			_, err := sut.kubeClient.CoreV1().Nodes().Create(context.Background(), &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+
+			nodeState1 := &sriovnetworkv1.SriovNetworkNodeState{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-node",
+					Generation: 123,
+				},
+			}
+			Expect(
+				createSriovNetworkNodeState(sut.client, nodeState1)).
+				To(BeNil())
+
+			nodeState2 := &sriovnetworkv1.SriovNetworkNodeState{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-node",
+					Generation: 777,
+				},
+			}
+			Expect(
+				updateSriovNetworkNodeState(sut.client, nodeState2)).
+				To(BeNil())
+
+			var msg Message
+			Eventually(refreshCh, "10s").Should(Receive(&msg))
+			Expect(msg.syncStatus).To(Equal("InProgress"))
+
+			Eventually(refreshCh, "10s").Should(Receive(&msg))
+			Expect(msg.syncStatus).To(Equal("Succeeded"))
+
+			Expect(sut.nodeState.GetGeneration()).To(BeNumerically("==", 777))
+		})
+	})
+})
+
+func TestConfigDaemon(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Config Daemon Suite")
+}
+
+func createSriovNetworkNodeState(c snclientset.Interface, nodeState *sriovnetworkv1.SriovNetworkNodeState) error {
+	_, err := c.SriovnetworkV1().
+		SriovNetworkNodeStates(namespace).
+		Create(context.Background(), nodeState, metav1.CreateOptions{})
+	return err
+}
+
+func updateSriovNetworkNodeState(c snclientset.Interface, nodeState *sriovnetworkv1.SriovNetworkNodeState) error {
+	_, err := c.SriovnetworkV1().
+		SriovNetworkNodeStates(namespace).
+		Update(context.Background(), nodeState, metav1.UpdateOptions{})
+	return err
+}
+
+type fakePlugin struct{}
```

Comment:
can we move this plugin maybe to the plugin folder?

with a comment that is a plugin for the unit tests

Reply:
sure